plugins/ssh-plugin/src/ssh-plugin-backend.ts

```diff
@@ -45,76 +54,93 @@ export async function start() {
     } else {
-        return Promise.resolve('');
+        fs.appendFileSync(configFile, keyConfig);
     }
 };
 
+const writeKey = function (name: string, key: string): void {
+    const keyFile = os.homedir() + '/.ssh/' + name.replace('.', '_');
+    fs.appendFileSync(keyFile, key);
+    fs.chmodSync(keyFile, '600');
+};
+
 const generateKeyPair = async function (sshkeyManager: SshKeyManager): Promise<void> {
-    const keyName = await theia.window.showInputBox({ placeHolder: 'Please provide a key pair name' });
-    const key = await sshkeyManager.generate(await getSshServiceName(), keyName ? keyName : '');
+    const keyName = `default-${Date.now()}`;
+    const key = await sshkeyManager.generate('vcs', keyName);
+    updateConfig(keyName);
+    writeKey(keyName, key.privateKey);
     const viewAction = 'View';
-    const action = await theia.window.showInformationMessage('Do you want to view the generated private key?', viewAction);
+    const action = await theia.window.showInformationMessage('Key pair successfully generated, do you want to view the public key?', viewAction);
     if (action === viewAction && key.privateKey) {
-        theia.workspace.openTextDocument({ content: key.privateKey });
+        const document = await theia.workspace.openTextDocument({ content: key.publicKey });
+        await theia.window.showTextDocument(document);
+    }
+};
+
+const generateKeyPairForHost = async function (sshkeyManager: SshKeyManager): Promise<void> {
+    const hostName = await getHostName();
+    const key = await sshkeyManager.generate('vcs', hostName);
+    updateConfig(hostName);
+    writeKey(hostName, key.privateKey);
+    const viewAction = 'View';
+    const action = await theia.window.showInformationMessage(`Key pair for ${hostName} successfully generated, do you want to view the public key?`, viewAction);
+    if (action === viewAction && key.privateKey) {
+        const document = await theia.workspace.openTextDocument({ content: key.publicKey });
+        await theia.window.showTextDocument(document);
     }
 };
 
 const createKeyPair = async function (sshkeyManager: SshKeyManager): Promise<void> {
-    const keyName = await theia.window.showInputBox({ placeHolder: 'Please provide a key pair name' });
+    const hostName = await getHostName();
     const publicKey = await theia.window.showInputBox({ placeHolder: 'Enter public key' });
+    const privateKey = await theia.window.showInputBox({ placeHolder: 'Enter private key' });
 
     await sshkeyManager
-        .create({ name: keyName ? keyName : '', service: await getSshServiceName(), publicKey: publicKey })
-        .then(() => {
-            theia.window.showInformationMessage('Key "' + `${keyName}` + '" successfully created');
+        .create({ name: hostName, service: 'vcs', publicKey: publicKey, privateKey })
+        .then(async () => {
+            theia.window.showInformationMessage('Key "' + `${hostName}` + '" successfully created');
```

Comment:
why do we use string literals and concatenation there ?

-->
`Key "${hostName}" successfully created`

Reply:
fixed